engine/crates/engine/src/registry/resolvers/mod.rs

```diff
@@ -235,11 +265,24 @@ impl Resolver {
                         variable_definitions,
                         registry,
                         Some(batcher),
-                    )
+                    );
+
+                #[cfg(feature = "tracing_worker")]
+                let future = future.instrument(info_span!("http_resolver"));
+
+                future
                     .await
                     .map_err(Into::into)
             }
-            Resolver::MongoResolver(resolver) => resolver.resolve(ctx, resolver_ctx).await.map_err(Into::into),
+            Resolver::MongoResolver(resolver) => {
+                let future = resolver.resolve(ctx, resolver_ctx);
+
+                #[cfg(feature = "tracing_worker")]
+                let future = future.instrument(info_span!("mongodb_resolver"));
```

Comment:
Could probably take some fields from `resolver` here as well (can't remember what they are for mongo off the top of my head, sure there are some though)